service/topology_coordinator.cc

```diff
@@ -0,0 +1,1911 @@
+                    });
+                    tlogger.trace("raft topology: CDC generation publisher fiber wakes up");
+                }
+            } catch (raft::request_aborted&) {
+                tlogger.debug("raft topology: CDC generation publisher fiber aborted");
+            } catch (seastar::abort_requested_exception) {
+                tlogger.debug("raft topology: CDC generation publisher fiber aborted");
+            } catch (group0_concurrent_modification&) {
+            } catch (term_changed_error&) {
+                tlogger.debug("raft topology: CDC generation publisher fiber notices term change {} -> {}", _term, _raft.get_current_term());
+            } catch (...) {
+                tlogger.error("raft topology: CDC generation publisher fiber got error {}", std::current_exception());
+                sleep = true;
+            }
+            if (sleep) {
+                try {
+                    co_await seastar::sleep_abortable(std::chrono::seconds(1), _as);
+                } catch (...) {
+                    tlogger.debug("raft topology: CDC generation publisher: sleep failed: {}", std::current_exception());
+                }
+            }
+            co_await coroutine::maybe_yield();
+        }
+    }
+
+    // Precondition: there is no node request and no ongoing topology transition
+    // (checked under the guard we're holding).
+    future<> handle_global_request(group0_guard guard) {
+        switch (_topo_sm._topology.global_request.value()) {
+        case global_topology_request::new_cdc_generation: {
+            tlogger.info("raft topology: new CDC generation requested");
+
+            auto tmptr = get_token_metadata_ptr();
+            auto [gen_uuid, guard_, mutation] = co_await prepare_and_broadcast_cdc_generation_data(tmptr, std::move(guard), std::nullopt);
+            guard = std::move(guard_);
+
+            topology_mutation_builder builder(guard.write_timestamp());
+            // We don't delete the request now, but only after the generation is committed. If we deleted
+            // the request now and received another new_cdc_generation request later, but before committing
+            // the new generation, the second request would also create a new generation. Deleting requests
+            // after the generation is committed prevents this from happening. The second request would have
+            // no effect - it would just overwrite the first request.
+            builder.set_transition_state(topology::transition_state::commit_cdc_generation)
+                   .set_new_cdc_generation_data_uuid(gen_uuid);
+            auto reason = ::format(
+                "insert CDC generation data (UUID: {})", gen_uuid);
+            co_await update_topology_state(std::move(guard), {std::move(mutation), builder.build()}, reason);
+            break;
+        }
+        case global_topology_request::keyspace_rf_change: {
```

Comment:
It's easy to miss this change, because it's in a newly added file, but here I'm actually adding handling for the newly introduced `global_topology_request::keyspace_rf_change`
So @tgrabiec you say (https://github.com/scylladb/scylladb/pull/16723#discussion_r1448027249 that I should create a new map of tablet replicas somewhere here?

Reply:
After redoing the refactor commit, I no longer have topology_coordinator.cc